File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    items = {}
    
    for c in clothes:
        if c[1] in items:
            items[c[1]] += 1
        else:
            items[c[1]] = 1
    
    for v in items.values():
        answer *= v + 1
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let answer = 1;
    const clothesType = {};
    
    for (const cloth of clothes) {
        if (!clothesType[cloth[1]]) {
            clothesType[cloth[1]] = 1;
        } else {
            clothesType[cloth[1]] += 1;
        }
    }
    
    for (const type in clothesType) {
        answer *= (clothesType[type] + 1);
    }
    
    return answer - 1;
}